### DISCUSSION TITLE
two contact sensor - Sonoff sv

### DISCUSSION THREAD

**Discussion by pecior** [RavenSystem/esp-homekit-devices#2021] [MEPLHAA Scripts]
I have a sonoff SV and I want to have two contractons on it, one on GPIO14 - GND the other GPIO5 - GND how to configure it ? Please help

**Comment by QDUNI** [comment]
What Homekit devices would you like to configure on the sonoff sv?

**Reply by pecior** [reply]
![sonoff sv](https://github.com/RavenSystem/esp-homekit-devices/assets/53180542/b9f058ea-2a76-4e33-aa70-15038613c07c)

**Reply by QDUNI** [reply]
Okay, so you want to configure a lock and door accessory ? 
Door to tell if its opened and lock to lock and unlock. 
Why do you need 2 magnetic reed switches ?

**Reply by pecior** [reply]
Yes
Yes
i want to use "sonoff sv" for home door (sensor)

**Reply by QDUNI** [reply]
I assume that the lock is able to open if you provide the 12 v to the lock. 

You only need 1 magnetic sensor. Place it one half on the door and the other on the door frame. 
If this sensor is open you know that the door isn't fully-closed and isnt locked either.
The lock is locked but the door isnt locked in the lock. 

This MEPLHAA creates 2 accesory's. 

- A lock which locks the lock if unlocked after 20 secs. 
- A Contact sensor to see if the door is opened or closed. 

```
{
  "c":{
    "io":[[[12],2],[[0],6 ,1],[[4,5,14],6,1]],
    "l":13,"b":[[0,5]]
  },
  "a":[{
    "t":4, // Lock with auto lock
    "i":20, // Return to locked state after 20 sec.
    "0":{"r":[[12,1,20]]}, // If unlocked action Power the lock for 20 sec. 
    "1":{"r":[[12]]} // Switch relay off
  },{
    "t":5, // Contact sensor accesory could be shown as door in homekit. 
    "0": { },
    "1": { },
    "g1": [ [ 4,0 ] ], // Indicate with magnetic sensor on gpio 4 if the door is closed or open
    "g0": [ [ 4,1 ] ]
  }
]
}

```
Compact form to use
```
{"c":{"io":[[[12],2],[[0],6,1],[[4,5,14],6,1]],"l":13,"b":[[0,5]]},"a":[{"t":4,"i":20,"0":{"r":[[12,1,20]]},"1":{"r":[[12]]}},{"t":5,"0":{},"1":{},"g1":[[4,0]],"g0":[[4,1]]}]}

**Reply by pecior** [reply]
my magnetic sensor is different. I changed to:

{"c":{"io":[[[12],2],[[0],6,1],[[4,5,14],6,1]],"l":13,"b":[[0,5]]},"a":[{"t":4,"i":20,"0":{"r":[[12,1,20]]},"1":{"r":[[12]]}},{"t":5,"0":{},"1":{},"g1":[[4,1]],"g0":[[4,0]]}]}

and it works. :) Thank you very much. :)

**Reply by QDUNI** [reply]
Great, no problem !
